src/VirtualClient/VirtualClient.Actions/SPECview/SpecViewExecutor.cs

```diff
@@ -0,0 +1,203 @@
+    {
+        private IFileSystem fileSystem;
+        private ISystemManagement systemManagement;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public SpecViewExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+            : base(dependencies, parameters)
+        {
+            this.fileSystem = dependencies.GetService<IFileSystem>();
+            this.systemManagement = dependencies.GetService<ISystemManagement>();
+        }
+
+        /// <summary>
+        /// The command line argument defined in the profile.
+        /// </summary>
+        public string CommandArguments
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(SpecViewExecutor.CommandArguments));
+            }
+        }
+
+        /// <summary>
+        /// The path to the RunViewperf.exe.
+        /// </summary>
+        public string ExecutablePath { get; set; }
+
+        /// <summary>
+        /// The path for the intermediate results file.
+        /// </summary>
+
+        /// <summary>
+        /// Defines the path to the SPECview package that contains the workload
+        /// executable.
+        /// </summary>
+        protected DependencyPath Package { get; set; }
+
+        /// <summary>
+        /// Executes the SPECview workload.
+        /// </summary>
+        protected override Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            return this.ExecuteWorkloadAsync(telemetryContext, cancellationToken);
+        }
+
+        /// <summary>
+        /// Initializes the environment
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            PlatformSpecifics.ThrowIfNotSupported(this.Platform);
+
+            await this.InitializePackageLocationAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            this.ExecutablePath = this.PlatformSpecifics.Combine(this.Package.Path, "RunViewperf.exe");
+        }
+
+        /// <summary>
+        /// Processes benchmark results
+        /// </summary>
+        private void CaptureMetrics(IProcessProxy workloadProcess, string commandArguments, EventContext telemetryContext)
+        {
+            if (workloadProcess.ExitCode == 0)
+            {
+                try
+                {
+                    // SPEC VIEW does not seem to support customized output folder. Results are outputted to a folder in the format of "results_20230913T052028"
+                    string[] subdirectories = this.fileSystem.Directory.GetDirectories(this.Package.Path, "results_*", SearchOption.TopDirectoryOnly);
```

Comment:
You might want to rename/delete a result file after parsing it. Otherwise second time you run it it will double parse that data.